tools-v2/pkg/cli/command/curvebs/delete/volume/clone_or_recover.go

```diff
@@ -0,0 +1,229 @@
+			fmt.Printf("UUID=%s, User=%s, Src=%s, Dest=%s\n", record.UUID, record.User, record.Src, record.File)
+		}
+	}
+
+	return records
+}
+
+func (c *CloneOrRecover) getCloneListAll(cloneType, status string) []Record {
+	limit := 100
+	offset := 0
+	var receiveRecords []Record
+
+	for {
+		records := c.getCloneList(cloneType, status, limit, offset)
+		if len(records) == 0 {
+			break
+		}
+		receiveRecords = append(receiveRecords, records...)
+		offset += limit
+	}
+
+	return receiveRecords
+}
+
+type QueryParams struct {
+	Action      string `json:"Action"`
+	Version     string `json:"Version"`
+	User        string `json:"User"`
+	UUID        string `json:"UUID"`
+	Source      string `json:"Source"`
+	Destination string `json:"Destination"`
+	Limit       string `json:"Limit"`
+	Offset      string `json:"Offset"`
+	Status      string `json:"Status"`
+	Type        string `json:"Type"`
+}
+
+func (c *CloneOrRecover) getCloneList(cloneType, status string, limit, offset int) []Record {
+	params := QueryParams{
+		Action:      "GetCloneTaskList",
+		Version:     version,
+		User:        c.User,
+		UUID:        c.TaskID,
+		Source:      c.Src,
+		Destination: c.Dest,
+		Limit:       strconv.Itoa(limit),
+		Offset:      strconv.Itoa(offset),
+		Status:      status,
+		Type:        cloneType,
+	}
+
+	var resp struct {
+		Code       string
+		Message    string
+		RequestId  string
+		TaskInfos  []Record
+		TotalCount int
+	}
+	err := c.query(params, &resp)
+	if err != nil || resp.Code != "0" {
+		fmt.Printf("get clone list fail, error=%s\n", err)
+		return nil
+	}
+
+	return resp.TaskInfos
+}
+
+func (c *CloneOrRecover) query(params QueryParams, data interface{}) error {
+	encodedParams := c.encodeParam(params)
+
+	subUri := fmt.Sprintf("/SnapshotCloneService?%s", encodedParams)
+
+	metric := basecmd.NewMetric(c.serverAddress, subUri, c.timeout)
+
+	result, err := basecmd.QueryMetric(metric)
+	if err.TypeCode() != cmderror.CODE_SUCCESS {
+		return err.ToError()
+	}
+
+	if err := json.Unmarshal([]byte(result), &data); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *CloneOrRecover) encodeParam(params QueryParams) string {
+	values := url.Values{}
+	paramsMap := map[string]string{
+		"Action":      params.Action,
+		"Version":     params.Version,
+		"User":        params.User,
+		"UUID":        params.UUID,
+		"Source":      params.Source,
+		"Destination": params.Destination,
+		"Limit":       params.Limit,
+		"Offset":      params.Offset,
+		"Status":      params.Status,
+		"Type":        params.Type,
+	}
+
+	for key, value := range paramsMap {
+		if value != "" {
+			values.Add(key, value)
+		}
+	}
+
+	return values.Encode()
+}
+
+func (c *CloneOrRecover) checkOK() bool {
+	var askRes string
+	fmt.Print("are you sure to clean(yes/no): ")
+	fmt.Scanln(&askRes)
+
+	if askRes == "yes" {
+		return true
+	} else if askRes == "no" {
+		return false
+	} else {
+		fmt.Println("Please enter yes or no")
+		return false
+	}
+}
```

Comment:
**Just in my opinion**. Sometimes the old tool checks the user's operation, but usually, the tool-v2 doesn't do that.